apps/ops/ws.py

```diff
@@ -71,8 +75,12 @@ async def receive_json(self, content, **kwargs):
         user = self.scope['user']
         user_role_ids = await self.get_current_user_role_ids(user)
         has_admin_auditor_role = bool(admin_auditor_role_ids & user_role_ids)
-
-        if not has_admin_auditor_role and task.name in self.user_tasks and task.creator != user:
+        has_perms = await self.has_perms(user, ['audits.view_joblog'])
+        # (有管理员或审计员角色) 或者 (任务是用户自己创建的 或者 有查看任务日志权限), 其他情况没有权限
+        if not (
+                has_admin_auditor_role or
+                (task.name in self.user_tasks and (task.creator == user or has_perms))
+        ):
             await self.send_json({'message': 'No permission', 'task': task_id})
             return
 
```

Comment:
I'm sorry, but the code you provided is incomplete. Can you please include all of the relevant code so I can take a look?